From aaea3e3f6c9b9fc8a4637eb2340fb080edc60acf Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Wed, 20 Nov 2024 14:04:19 +0100
Subject: [PATCH 1353/1695] plat-stm32mp2: use software PRNG when
 CFG_STM32_CM33TDCID is enabled

RNG peripheral is allocated to M33, so disable HWRNG_PTA and enable
software PRNG.
software PRNG cannot be seeded by RNG peripheral, for now we had to use
the OP-TEE default seeding procedure.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 3bd3aa98acf608d88475d6126aa32c588407c22e plat-stm32mp2: use software PRNG when CFG_STM32_CM33TDCID is enabled]
Change-Id: I2f8d0bf5e75ebd660a191e9835d8a8af292b1133
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/419774
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/arch/arm/plat-stm32mp2/conf.mk | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index 339e3a991..53fdcbb16 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -86,6 +86,8 @@ endif
 
 CFG_STM32_CM33TDCID ?= n
 ifeq ($(CFG_STM32_CM33TDCID),y)
+$(call force,CFG_HWRNG_PTA,n)
+CFG_WITH_SOFTWARE_PRNG ?= y
 $(call force,CFG_SCMI_CORTEXM_AGENT,n)
 $(call force,CFG_SCMI_MSG_DRIVERS,n)
 $(call force,CFG_SCMI_PTA,n)
-- 
2.47.2

